tools/iree-minimal-invoke-module-main.c

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// Minimal C program linking in the IREE runtime. Does not even take enough
+// command-line parameters to be able to successfully run most modules. Only
+// useful for code size/symbols monitoring purposes.
+
+#include <stdio.h>
+
+// Low-level IREE VM APIs.
+// The higher-level iree/runtime/api.h can be used for more complete ML-like
+// programs using the hardware abstraction layer (HAL). This simple sample just
+// uses base VM types.
+#include "iree/base/api.h"
+#include "iree/base/internal/file_io.h"
+#include "iree/modules/hal/types.h"
+#include "iree/tooling/context_util.h"
+#include "iree/vm/api.h"
+#include "iree/vm/bytecode_module.h"
+
+// NOTE: CHECKs are dangerous but this is a sample; a real application would
+// want to handle errors gracefully. We know in this constrained case that
+// these won't fail unless something is catastrophically wrong (out of memory,
+// solar flares, etc).
+int main(int argc, char** argv) {
+  if (argc != 3) {
+    fprintf(stderr,
+            "Usage:\n"
+            "  %s </path/to/say_hello.vmfb> <entry.point>\n",
+            argv[0]);
+    fprintf(
+        stderr,
+        "This program performs a dummy invoke without actual parameters, so "
+        "the function call will fail in general. It is only meant to be used "
+        "to inspect IREE runtime code size and internal call graph.\n");
+    return -1;
+  }
+
+  // Internally IREE does not (in general) use malloc and instead uses the
+  // provided allocator to allocate and free memory. Applications can integrate
+  // their own allocator as-needed.
+  iree_allocator_t allocator = iree_allocator_system();
+
+  // Create the root isolated VM instance that we can create contexts within.
+  iree_vm_instance_t* instance = NULL;
+  IREE_CHECK_OK(iree_vm_instance_create(allocator, &instance));
+  IREE_CHECK_OK(iree_hal_module_register_all_types(instance));
+
+  // Load the module from stdin or a file on disk.
+  // Applications can ship and load modules however they want (such as mapping
+  // them into memory instead of allocating like this). Modules can also be
+  // embedded in the binary but in those cases it makes more sense to use emitc
+  // to avoid the bytecode entirely and have a fully static build (see
+  // samples/emitc_modules/ for some examples).
```

Comment:
We can omit some of these comments from a program authored purely for size monitoring ("see some/path/to/main.c for a more fully featured sample application with comments explaining the APIs used")